[JENKINS-61224] Removing the "monitor[s] text next to the bell (and color change)

romenrg opened the pull request:
### Description
See [JENKINS-61224](https://issues.jenkins-ci.org/browse/JENKINS-61224).

This PR removes the "monitor[s]" text next to the bell, since as discussed, seems unnecessary and a bit confusing. Other tools (e.g. GitHub) only have the bell and a circle.

EDIT: It also changes the colors of the "monitor" (alert). See comments below.

#### Screenshots
Before on computer:
<img width="1440" alt="before-bell-no-text-computer" src="https://user-images.githubusercontent.com/1831480/76164985-26d29d80-614b-11ea-9099-e48da90779df.png">

After on computer:
<img width="1440" alt="after-bell-no-text-computer" src="https://user-images.githubusercontent.com/1831480/76164987-2df9ab80-614b-11ea-96e7-bc671a576308.png">

Before on mobile (already not showing due to responsive behavior):
<img width="248" alt="before-bell-no-text-mobile" src="https://user-images.githubusercontent.com/1831480/76164990-34882300-614b-11ea-9d4e-caf05caae2de.png">

After on mobile (no change):
<img width="247" alt="after-bell-no-text-mobile" src="https://user-images.githubusercontent.com/1831480/76165005-49fd4d00-614b-11ea-8aef-aa3a61732123.png">

### Proposed changelog entries

* Remove the "monitor[s]" text next to the bell, as unnecessary, leading to a cleaner UI
* Change the colors of the notifications next to the bell, to make them more noticeable

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] JIRA issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [X] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@fqueiruga 
@batmat 
@jglick
@uhafner 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

romenrg committed:
[JENKINS-61224] Removed span tag including the "monitor[s]" text

romenrg committed:
[JENKINS-61224] Removed monitors from the i18n properties files

batmat commented:
@romenrg very cool you were able to work on this!
Could you please provide a screenshot showing the state before/after the change in the description?

Thanks!

romenrg commented:
@batmat Sure! I've edited the description to add the screenshots.
Very happy to contribute my two cents here :)

markjacksonfishing reviewed (approved):
I tested this and it looks good. Thank you @romenrg

daniel-beck reviewed (changes requested):
There are many occurrences of the term left on the UI (e.g. `AdministrativeMonitorsConfiguration` UI), and only addressing this part won't actually resolve the linked issue.

A more exhaustive search should be done, and a new term established, where we cannot go without a label.

daniel-beck commented:
FWIW I provided the feedback https://github.com/jenkinsci/jenkins/pull/4463#issuecomment-587516396 to the original change in which I say it's OK to make the UI "less red" because the additional width compensates for it. Now we're not red, and not wide.

romenrg commented:
@daniel-beck To your comments, some thoughts:
- Regarding the occurrences of the term in other places and potential renaming, I think that's a separate topic. I would say this is purely a UI decision, in which we are deciding not to show the text "monitor[s]", but that doesn't imply we are renaming the concept, so I would say that, as of now, the concept remains as "monitor", we have just decided to remove the work from the UI
- Regarding your other comment about the colors, maybe also a separate concern, but if you fee removing the text further de-emphasizes the importance of the notification, I would be happy to invert the colors of the background and the text to make it more noticeable, if you people agree. See the following screenshot of this alternative proposal of the button:
<img width="1435" alt="Inverting-colors-notificaiton" src="https://user-images.githubusercontent.com/1831480/76166362-d44aae80-6155-11ea-9cc0-e093bfc1eb18.png">
(Actually I like this alternative better, red background, white number, as traditionally, but keeping the new rounded shape)

daniel-beck commented:
@romenrg 

> Regarding your other comment about the colors, maybe also a separate concern?

I mean – not really, both size and color can be used to make the element noticeable. First we replaced color with size, now we take away size.

I like your alternative suggestion a lot. Thanks for that. Let's hope the more design inclined among reviewers agree. 👍 

> Regarding the occurrences of the term in other places and potential renaming, I think that's a separate topic

Unsure. Can probably go either way. Why I think these are connected: While the linked issue calls out that the term is used elsewhere already and calls out that use as _more tolerable_, the problem you're addressing here is identified as the term being misleading ("not intuitive in this context") to users, rather than e.g. an aesthetic choice. But if it's misleading here, it's misleading everywhere.

uhafner commented:
I like the change 👍 I looks much better now.

romenrg commented:
@uhafner Thanks for reviewing :). Just to clarify, the change you like is the [original one](https://github.com/jenkinsci/jenkins/pull/4555#issue-385273124) in which I just removed the text and / or [the new colors](https://github.com/jenkinsci/jenkins/pull/4555#issuecomment-596220942) I suggested after discussing with @daniel-beck?

uhafner commented:
Ah, sorry. Indeed there are two different changes. My intention was to say that I like the badge  without text (I do not use this monitor anyway so for me it just removes some clutter from the layout). 

About the colors: Here I actually would prefer the red background and white text as this is the typical indication of an alert. The current design somehow does not indicate that this should be  an important warning.

fqueiruga commented:
I say if we are going to go full red background, we should do it right. The current red / white background has a rather bad contrast value (https://contrast-ratio.com/#%23fff-on-%23F94B4E) and it looks even worse agains a non-black header background (like with the ui refresh flag enabled).

I'd say let's explore some options with a better text contrast. One example are bootstrap alerts colors. They offer a great contrast in text against  background (https://contrast-ratio.com/#%23721c24-on-%23f8d7da):

<img width="438" alt="Captura de pantalla 2020-03-09 a las 10 51 55" src="https://user-images.githubusercontent.com/5738588/76202145-abc5c180-61f4-11ea-8616-b98a4e954dab.png">

<img width="442" alt="Captura de pantalla 2020-03-09 a las 10 52 08" src="https://user-images.githubusercontent.com/5738588/76202153-aec0b200-61f4-11ea-83f5-2f2f7114c5fd.png">

These are way easier to read. FTR I'm against choosing the bootstrap colors, but we can do some research and find a color combination that works for Jenkins. WDYT?

romenrg commented:
@fqueiruga Good point about the constrast.

I do like your proposal of using the [bootstrap "alert-danger"](https://getbootstrap.com/docs/4.4/components/alerts/#link-color) color and background-color for the new UI.

However, for the old UI I think that soft red background is too subtle. I been looking at bootstrap and I think the colors of the ["badge-danger"](https://getbootstrap.com/docs/4.4/components/badge/#pill-badges) are more easy to notice and have a [better contrast](https://contrast-ratio.com/#%23fff-on-%23dc3545) than the original suggestion (not perfect but AA for all font sizes and AAA for "bold above 14pt" which is seems to be the case for this notification. Screenshot:
<img width="240" alt="badge-danger-color" src="https://user-images.githubusercontent.com/1831480/76208215-255eaf00-61f7-11ea-909f-b8186e642a31.png">

I think it is reasonable to have different colors for the notification in the old UI and in the new UI, since the black vs blue background of the navbar has an impact on how the notification is perceived. What do you people think?

daniel-beck commented:
I think @romenrg makes good points in his last comment. A slightly different shade of red would have the contrast we want IIUC.

Wadeck commented:
Two cents on colors, the bootstrap alert-danger colors are nice on white background, but both of our style are completely different from white.

I really like the badge-danger approach Romen proposed for the black one.

romenrg committed:
[JENKINS-61224] Different colors for monitor count in old vs new UI

Both sets of colors provide good contrast. It now sets a "stronger" red background with white text for the old UI (makes it more noticeable in the black navbar) and a softer red with a darker text for the new UI (looks better with the blue bg). Both come from boostrap styles ("badge-danger" vs "alert-danger")

romenrg commented:
I've gone ahead and pushed the color combination that seems to have obtained the broader agreement. The result is now the following:

For the new UI, we are using the bootstrap "alert-danger" set of colors, as proposed by @fqueiruga. See screenshot:
<img width="219" alt="bell-no-text-color-newUI" src="https://user-images.githubusercontent.com/1831480/76236496-ac764c00-6224-11ea-806f-d8c0eefd73c5.png">

While for the old UI, we are using the bootstrap "badge-danger" combination of colors, as I proposed to address @daniel-beck 's concern about the "monitor" not receiving enough attention. See screenshot:
<img width="257" alt="bell-no-text-color-oldUI" src="https://user-images.githubusercontent.com/1831480/76236586-cfa0fb80-6224-11ea-849c-3cd890aa9159.png">

I'm quite happy with the result. I think the combination of removing the text and using these new sets of colors is quite nice, both in the old and in the new UI :)

romenrg committed:
Merge remote-tracking branch 'upstream/master' into bell-no-text

romenrg committed:
[JENKINS-61224] Trivial commit on comment to force re-build

uhafner commented:
- The badge-danger red for the old layout 👍 
- The alert-danger red for the new layout 👎: it is not part of the new color palette and seems not to fit to the blue color in my eyes

But anyway: I hope that both colors are part of theming so they can be replaced with the matching colors of a theme

daniel-beck commented:
Unsure about the refreshed UI, but the classic one is a lot better now 👍 

> ![Screenshot 2020-03-11 at 06 51 48](https://user-images.githubusercontent.com/1831569/76386528-02eaa400-6365-11ea-8162-94996fe4abf9.png)
> ![Screenshot 2020-03-11 at 06 51 58](https://user-images.githubusercontent.com/1831569/76386531-067e2b00-6365-11ea-829f-3dec8aad5201.png)
> ![Screenshot 2020-03-11 at 06 52 51](https://user-images.githubusercontent.com/1831569/76386532-0847ee80-6365-11ea-8a25-fc99a8870a7d.png)
> ![Screenshot 2020-03-11 at 06 53 04](https://user-images.githubusercontent.com/1831569/76386538-09791b80-6365-11ea-8def-57159350e9f0.png)

daniel-beck commented:
I played around with a red background on the refreshed UI, but it has the usual blue/red problem of looking messed up (TIL that is called chromostereopsis). So that probably needs a different approach.

batmat commented:
@josephbrueggen can you please chime in on the color palette issue raised by Ulli? Thanks

romenrg committed:
Merge branch 'master' into bell-no-text

oleg-nenashev reviewed (approved):
It looks good to me, thanks! Putting on hold as per @batmat's review request

batmat commented:
@uhafner in the meantime, I would love if you can clarify your stance: are you actually requesting a change here? If so can you please do it so using the GitHub feature 
![image](https://user-images.githubusercontent.com/223853/76505724-3b17e280-644a-11ea-99eb-d644960d0911.png)

If you're actually fine with this change, but would mainly like us to double-check this even after a merge, that would be good to know too. 
Danke!

fqueiruga commented on war/src/main/less/abstracts/colors.less:
```diff
@@ -28,5 +28,9 @@
 @breadcrumbs-border: #EAEFF2;
 
+// Monitor / alert new colors
+@monitor-bg: #f8d7da;
```
```suggestion
@monitor-bg-v2: #f8d7da;
```

fqueiruga commented on war/src/main/less/abstracts/colors.less:
```diff
@@ -28,5 +28,9 @@
 
+// Monitor / alert new colors
+@monitor-bg: #f8d7da;
+@monitor-color: #721c24;
```
```suggestion
@monitor-color-v2: #721c24;
```

fqueiruga commented on war/src/main/less/ui-refresh-overrides.less:
```diff
@@ -82,4 +82,9 @@
     }
+
+    #visible-am-button .am-monitor__count {
+        color: @monitor-color;
```
```suggestion
        color: @monitor-color-v2;
```

fqueiruga commented on war/src/main/less/ui-refresh-overrides.less:
```diff
@@ -82,4 +82,9 @@
+
+    #visible-am-button .am-monitor__count {
+        color: @monitor-color;
+        background-color: @monitor-bg;
```
```suggestion
        background-color-v2: @monitor-bg;
```

fqueiruga reviewed (approved):
These changes are suggestions to keep variable naming consistent

romenrg committed:
[JENKINS-61224] Adding the "-v2" suffix to new UI vars

uhafner commented:
No, I do not request a change here (we can change some colors later on if required).

oleg-nenashev commented:
I will merge it in 24 hours if no negative feedback.
Thanks @romenrg !

josephbrueggen commented:
Thanks @romenrg for the contribution here and also @uhafner for being mindful of the color palette. In the future we may return to address color consistency, but from my perspective this is ready to merge. If we do return to make a minor color adjustment, we will prioritize the following:

- sufficient visual contrast _and_
- ensuring a sufficient sense of urgency since these warnings are often critical in nature

Regardless, let's merge this—thanks everyone!

romenrg commented:
Thanks everyone for the discussions and feedback. Happy to see this merged now :)

BTW, I've just edited the PR description to add a second "proposed changelog entry" about the color change that we ended up introducing here. Not sure if the changelog entries are processed manually or automatically, so I'm adding this comment to let you know about it.
